src/lib/swapper/swappers/ThorchainSwapper/utils/getL1ToLongtailQuote.ts

```diff
@@ -0,0 +1,174 @@
+import { convertDecimalPercentageToBasisPoints } from 'state/slices/tradeQuoteSlice/utils'
+
+import type { ThorTradeQuote } from '../getThorTradeQuote/getTradeQuote'
+import { addAggregatorAndDestinationToMemo } from './addAggregatorAndDestinationToMemo'
+import { getBestAggregator } from './getBestAggregator'
+import { getL1quote } from './getL1quote'
+import type { AggregatorContract } from './longTailHelpers'
+import { getTokenFromAsset, getWrappedToken, TradeType } from './longTailHelpers'
+
+// This just gets uses UniswapV3 to get the longtail quote for now.
+export const getL1ToLongtailQuote = async (
+  input: GetTradeQuoteInput,
+  streamingInterval: number,
+  assetsById: AssetsByIdPartial,
+): Promise<Result<ThorTradeQuote[], SwapErrorRight>> => {
+  const {
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
+    sellAsset,
+    slippageTolerancePercentageDecimal,
+  } = input
+
+  /*
+    We only support L1 -> ethereum longtail swaps for now.
+  */
+  if (buyAsset.chainId !== ethChainId) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - Unsupported chainId ${buyAsset.chainId}.`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { buyAssetChainId: buyAsset.chainId },
+      }),
+    )
+  }
+
+  const chainAdapterManager = getChainAdapterManager()
+  const sellChainId = sellAsset.chainId
+  const buyChainId = buyAsset.chainId
+
+  const nativeSellAssetId = chainAdapterManager.get(sellChainId)?.getFeeAssetId()
+  const nativeSellAsset = nativeSellAssetId ? assetsById[nativeSellAssetId] : undefined
+
+  const nativeBuyAssetId = chainAdapterManager.get(buyChainId)?.getFeeAssetId()
+  const nativeBuyAsset = nativeBuyAssetId ? assetsById[nativeBuyAssetId] : undefined
+
+  if (!nativeBuyAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${buyChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { buyAssetChainId: buyChainId },
+      }),
+    )
+  }
+
+  if (!nativeSellAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${sellChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { sellAssetChainId: sellChainId },
+      }),
+    )
+  }
+
+  // TODO: Move these constants outside
+  // TODO2: use more than just UniswapV3, and also consider trianglar routes.
+  const ALLOWANCE_CONTRACT = '0xF892Fef9dA200d9E84c9b0647ecFF0F34633aBe8' // TSAggregatorTokenTransferProxy
+
+  const l1Tol1QuoteInput: GetTradeQuoteInput = {
+    ...input,
+    buyAsset: nativeBuyAsset,
+    sellAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
+  }
+
+  const thorchainQuotes = await getL1quote(
+    l1Tol1QuoteInput,
+    streamingInterval,
+    TradeType.L1ToLongTail,
+  )
+
+  const unwrappedThorchainQuotes = thorchainQuotes.unwrap()
```

Comment:
We probably want to check for `isErr()` before unwrapping as this will otherwise throw